brave/brave-core
# Tor profile service

fix https://github.com/brave/brave-browser/issues/361
fix https://github.com/brave/brave-browser/issues/360
fix https://github.com/brave/brave-browser/issues/376
## Submitter Checklist:


## Test Plan:
- Manual test
   1. Open Tor window and check tor connectivity by going to https://check.torproject.org/ <img width="381" alt="screen shot 2018-09-12 at 19 44 21" src="https://user-images.githubusercontent.com/11330831/45464262-a2719c80-b6c4-11e8-9927-2cb4192e81cf.png">
   2. Use fresh profile and "Open Tor Window" will show up only when tor executable is downloaded (see cldoidikboihgcjfkhdeidbpclkineef in user dir)
   3.  When https://check.torproject.org/ shows you are connected to tor network, click "New Tor Identity" in app menu and page will refresh and shows different ip address <img width="440" alt="screen shot 2018-09-12 at 19 51 25" src="https://user-images.githubusercontent.com/11330831/45464454-7d315e00-b6c5-11e8-8e6c-458f26eb2d4a.png">
   4.  Quit Brave and do `ps aux |grep cldoidikboihgcjfkhdeidbpclkineef`, there shouldn't be any tor 
process left
   5. Shield should work in both tor window and guest window

- Unit test
  1. BraveProfileManager
  2. BraveTorNetworkDelegateHelper 

## Reviewer Checklist:
